src/common/copilot/dataverseMetadata.ts

```diff
@@ -220,3 +221,104 @@ function parseYamlContent(content: string, telemetry: ITelemetry, sessionID: str
     }
 }
 
+export async function getDefaultLanguageCode(orgUrl:string, telemetry: ITelemetry, sessionID: string, dataverseToken: string) {
+    let defaultPortalLanguageCode = vscode.env.language;
+    if (IS_DESKTOP) {
+        const lcid = await fetchLanguageCodeId();
+        defaultPortalLanguageCode = await fetchLanguageCodeFromAPI(orgUrl, dataverseToken, telemetry, sessionID, lcid);
+    } else {
+        defaultPortalLanguageCode = getDefaultLanguageCodeWeb();
+    }
+    return defaultPortalLanguageCode;
+}
+
+async function readWebsiteYAML(filePath: string): Promise<string | null> {
+    const workspaceFolder = vscode.workspace.getWorkspaceFolder(
+        vscode.Uri.file(filePath)
+    );
+
+    if (workspaceFolder) {
+        const workspaceFolderPath = workspaceFolder.uri.fsPath;
+        return await findWebsiteYAML(filePath, workspaceFolderPath);
+    }
+
+    return null;
+}
+
+async function fetchLanguageCodeId(): Promise<string> {
+    try {
+        let activeFilePath = "";
+        if (vscode.window.activeTextEditor) {
+            activeFilePath = vscode.window.activeTextEditor.document.uri.fsPath;
+        } else if (vscode.workspace.workspaceFolders?.length === 1) {
+            activeFilePath = vscode.workspace.workspaceFolders[0].uri.fsPath;
+        } else {
+            // Handle multiple workspace folders when no active text editor is present
+            return "";
+        }
+
+        const yamlContent = await readWebsiteYAML(activeFilePath);
+        if (yamlContent) {
+            const parsedYAML = yaml.parse(yamlContent);
+            const languageCodeId = parsedYAML[EntityMetadataKeyAdx.LANGUAGE_ID];
+            return languageCodeId;
+        } else {
+            return "";
+        }
+    } catch (error) {
+        return "";
+    }
+}
+
+async function fetchLanguageCodeFromAPI(
+    orgUrl: string,
+    apiToken: string,
+    telemetry: ITelemetry,
+    sessionID: string,
+    lcid: string
+): Promise<string> {
+    try {
+        const dataverseApiUrl = `${
```

Comment:
No, so downloaded content has lcid, but from lcid we have to pass language code to copilot, for localization. We could have created a map as constant for all the languages we support for now, but this can be a better solution keeping in mind in future we can support more languages.